go/marshal/encode.go

```diff
@@ -332,3 +336,27 @@ func listEncoder(t reflect.Type, parentStructTypes []reflect.Type) encoderFunc {
 	elemEncoder = typeEncoder(t.Elem(), parentStructTypes)
 	return e
 }
+
+func mapEncoder(t reflect.Type, parentStructTypes []reflect.Type) encoderFunc {
+	e := encoderCache.get(t)
+	if e != nil {
+		return e
+	}
+
+	var keyEncoder encoderFunc
+	var valueEncoder encoderFunc
+	e = func(v reflect.Value) types.Value {
+		keys := v.MapKeys()
+		kvs := make([]types.Value, 2*len(keys))
+		for i, k := range keys {
+			kvs[2*i] = keyEncoder(k)
+			kvs[2*i+1] = valueEncoder(v.MapIndex(k))
```

Comment:
I find myself not doing that because append allocates a new object every time (the backing store is kept)